### DISCUSSION TITLE
The new Synthesizer class about generating synthetic data

### DISCUSSION THREAD

**Discussion by nshutijean** [stanfordnlp/dspy#596] [Q&A]
I am trying to play around with the new synthetic data generation feature but I couldn't load the Synthesizer class from `from dspy.datasets import DataLoader, Synthesizer`  which I think was merged. Using the example below

```
from dspy.datasets import DataLoader, Synthesizer

dl = DataLoader()

data = dl.from_huggingface(
    "gsm8k", "main",
    fields=("question", "answer"),
    input_keys=("question",),
    split="train[:3]"
)

synthesizer = Synthesizer()

syn_data = synthesizer.generate(examples=data, num_data=10)

synthesizer.export(data=syn_data, path="syn_data_gpt_4.csv")
```
Also any other help, clarification would be very much appreciated

**Comment by taylor-shift** [comment]
https://colab.research.google.com/drive/1CweVOu0qhTC0yOfW5QkLDRIKuAuWJKEr?usp=sharing

**Reply by taylor-shift** [reply]
This seems to work for me...
